### DISCUSSION TITLE
Parameters for rescoring peptidomics data

### DISCUSSION THREAD

**Discussion by Dmorgen** [CompOmics/ms2rescore#166] [Q&A] [answered]
Hi,

Where can I change the enzyme parameters in the GUI application for peptidomics data? I've tried to input in the Mokapot FASTA parameters the Enzyme as '.' (as written in the Mokapot documentation) but I got an error that the Enzyme parameters are invalid. Is there anything I should do differently?

Thanks!
David.

**Comment by MatteoLacki** [comment]
Same question. Experimented with:

```
            "mokapot": {
                "write_flashlfq": true,
                "write_txt": true,
                "write_weights": true,
                "enzyme":"[KR](?!P)"
            }
```
and 
```
"rescoring_engine": {
            "mokapot": {
                "write_weights": true,
                "write_txt": true,
                "write_flashlfq": false,
                "protein_kwargs": {
                    "enzyme":"[KR](?!P)"
                }
            }
        },
```
and it all fails. so it's back to the source and that's not simple to parse.

**Comment by MatteoLacki** [comment]
I manage to get it work: one needs  to provide both fasta and protein_kwargs:
```json
"rescoring_engine": {
            "mokapot": {
                "write_flashlfq": true,
                "write_txt": true,
                "write_weights": true,
                "fasta_file": "my_favorit_fasta.fasta",
                "protein_kwargs": {
                    "enzyme":"[KR](?!P)"
                }
            }
        },
```

**Comment by MatteoLacki** [comment]
Looking retrospectively: this is likely maybe not a bug, but an inconsistency: when I was running the command line tool, I did pass in the `-f` command. It was not passed on further to `mokapot`. I will make it into an enhancement report.

**Comment by RalfG** [comment, marked as answer]
Hi @Dmorgen,

Thanks for your interest in MS²Rescore! 

Like Matteo mentioned, to enable protein inference and protein-level confidence estimation in Mokapot through MS²Rescore, you must provide both `fasta_file` and `protein_kwargs` options. These are passed to the [mokapot.read_fasta](https://mokapot.readthedocs.io/en/latest/api/functions.html#mokapot.read_fasta) function.

For a non-specific digest, such as in immunopeptidomics, the settings might look like this:

```
"rescoring_engine": {
    "mokapot": {
        "fasta": "uniprot-proteome-human-contaminants.fasta",
        "protein_kwargs": {
            "enzyme": ".",
            "missed_cleavages": 16,
            "max_length": 16,
            "min_length": 7,
            "semi": false
        }
    }
},
```

Note that in this case, the missed cleavages should be set to the same number as `max_length`, although `0` might also work (could not find that in the Mokapot docs).

One caveat here is that depending on the size of the search space (amount and length of proteins), Mokapot might take a very long time to calculate the full digest of proteins into peptides, and protein inference on immunopeptidomics experiments is perhaps also not what you are looking for.

Depending on the MS²Rescore PSM input file format (coming from the search engine), the mapping from peptides to proteins will already be there, and will be propagated to the PSMs TSV MS²Rescore output file.

Best,
Ralf

**Comment by Dmorgen** [comment]
Thank you Ralf and Mateo!
